cranelift/codegen/src/egraph.rs

```diff
@@ -53,7 +51,7 @@ pub struct EgraphPass<'a> {
     /// "Domtree with children": like `domtree`, but with an explicit
     /// list of children, complementing the parent pointers stored
     /// in `domtree`.
-    domtree_children: DomTreeWithChildren,
+    domtree_children: DominatorTreePreorder,
```

Comment:
Can we update the name and comment here to match the type ("a preorder iterator over the domtree" or something similar)?

Reply:
When I looked again today at these modules to decide what to rename the fields to and what documentation to write, I discovered that the original `DominatorTree` isn't actually needed anywhere once we have a `DominatorTreePreorder` available. There was one `debug_assert!` using the original tree for dominance queries, but we can do those in O(1) time against the preorder instead. So I've renamed `domtree_children` just to `domtree`, replacing the other `domtree` field entirely.

Reply:
Nice!